enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -176,6 +176,30 @@ func openFile(ctx context.Context, env environment.Env, fileName string) (io.Rea
 	return env.GetFileResolver().Open(fileName)
 }
 
+// Returns the cgroup version available on this machine, which is returned from
+//
+//	stat -fc %T /sys/fs/cgroup/
+//
+// and should be either 'cgroup2fs' (version 2) or 'tmpfs' (version 1)
+//
+// More info here: https://kubernetes.io/docs/concepts/architecture/cgroups/
+func getCgroupVersion() string {
+	b, err := exec.Command("stat", "-fc", "%T", "/sys/fs/cgroup/").Output()
+	if err != nil {
+		log.Debugf("Error determining system cgroup version. Falling back to --executor.firecracker_cgroup_version (which is \"%s\"). Error: %s", *firecrackerCgroupVersion, err.Error())
+		return *firecrackerCgroupVersion
```

Comment:
Instead of falling back to this flag, I think it would be better to change the semantics of the flag so that an empty value (new default) means auto detect and any other value is used explicitly. If the flag value is "" and we cannot auto-detect then we return error.

It would also be nice if we could detect this earlier rather than when a container is being created. In the podman implementation we have an abstraction where we create a provider on startup and the provider in turns creates new container instances. This abstraction gives us a place to do startup configuration like this and be able to return error conditions.

Reply:
Yeah, that all sounds good. I added a TODO for the second part because it's a little bit more involved, LMK what you think.